src/states/elder/mod.rs

```diff
@@ -2090,7 +2100,16 @@ impl Approved for Elder {
             self.remove_from_proving_section_cache(&sec_info);
             // Vote for neighbour update if we haven't done so already.
             // vote_for_event is expected to only generate a new vote if required.
-            self.vote_for_event(sec_info.into_network_event());
+            self.vote_for_event(sec_info.clone().into_network_event());
+        }
+
+        if !self_sec_update {
+            // Sending out Ack message.
+            let src = Authority::Section(*self.name());
+            let dst = Authority::Section(sec_info.prefix().name());
+            let content = MessageContent::SectionInfoAck(sec_info);
```

Comment:
I would probably stick to `AckMessage` as in the issue unless there is a good reason.